redis/sentinel.py

```diff
@@ -125,16 +120,36 @@ def rotate_slaves(self):
             pass
         raise SlaveNotFoundError('No slave found for %r' % (self.service_name))
 
-    def _checkpid(self):
```

Comment:
The implementation in `ConnectionPool` is thread-safe and the `__init__()` call is not needed, `reset()` does everything necessary.